File: sdk/python/cmd/pulumi-language-python/testdata/toml/published/projects/l3-map-keys/myComponent.py
import pulumi
from pulumi import Input
from typing import Optional, Dict, TypedDict, Any
import pulumi_primitive as primitive

class MyComponentArgs(TypedDict):
    booleanMap: Input[Dict[str, bool]]

class MyComponent(pulumi.ComponentResource):
    def __init__(self, name: str, args: MyComponentArgs, opts:Optional[pulumi.ResourceOptions] = None):
        super().__init__("components:index:MyComponent", name, args, opts)

        res = primitive.Resource(f"{name}-res",
            boolean=False,
            float=2.17,
            integer=-12,
            string="adversarial",
            number_array=[
                float(0),
                float(1),
            ],
            boolean_map=args["booleanMap"],
            opts = pulumi.ResourceOptions(parent=self))

        self.boolean_map = res.boolean_map
        self.register_outputs({
            'booleanMap': res.boolean_map
        })

File: sdk/nodejs/cmd/pulumi-language-nodejs/testdata/published/bun/projects/l3-map-keys/myComponent.ts
import * as pulumi from "@pulumi/pulumi";
import * as primitive from "@pulumi/primitive";

interface MyComponentArgs {
    booleanMap: pulumi.Input<Record<string, pulumi.Input<boolean>>>,
}

export class MyComponent extends pulumi.ComponentResource {
    public booleanMap: pulumi.Output<Record<string, pulumi.Input<boolean>>>;
    constructor(name: string, args: MyComponentArgs, opts?: pulumi.ComponentResourceOptions) {
        super("components:index:MyComponent", name, args, opts);
        const res = new primitive.Resource(`${name}-res`, {
            boolean: false,
            float: 2.17,
            integer: -12,
            string: "adversarial",
            numberArray: [
                0,
                1,
            ],
            booleanMap: args.booleanMap,
        }, {
            parent: this,
        });

        this.booleanMap = res.booleanMap;
        this.registerOutputs({
            booleanMap: res.booleanMap,
        });
    }
}
